tests/tracking/test_rest_tracking.py

```diff
@@ -604,8 +606,8 @@ def assert_bad_request(payload, expected_error_message):
 def test_log_model(mlflow_client):
     experiment_id = mlflow_client.create_experiment("Log models")
     with TempDir(chdr=True):
-        mlflow.set_experiment("Log models")
         model_paths = ["model/path/{}".format(i) for i in range(3)]
+        mlflow.set_tracking_uri(mlflow_client.tracking_uri)
```

Comment:
Needs to be set so that the run context has reference to the created experiment in the patched mlflow_client obj